supervisor/store/types.py

```diff
@@ -0,0 +1,26 @@
+"""Repository type definitions for the store."""
+
+from enum import StrEnum
+
+from ..const import REPOSITORY_CORE, REPOSITORY_LOCAL
+
+
+class SpecialRepository(StrEnum):
+    """Special built-in repositories with unique behavior."""
+
+    CORE = REPOSITORY_CORE
+    LOCAL = REPOSITORY_LOCAL
+
+
+class DefaultRepository(StrEnum):
+    """Default URL-based repositories that come pre-configured."""
+
+    COMMUNITY_ADDONS = "https://github.com/hassio-addons/repository"
+    ESPHOME = "https://github.com/esphome/home-assistant-addon"
+    MUSIC_ASSISTANT = "https://github.com/music-assistant/home-assistant-addon"
+
+
+# All repositories that are considered "built-in" (special + default)
+ALL_BUILTIN_REPOSITORIES = {repo.value for repo in SpecialRepository} | {
+    repo.value for repo in DefaultRepository
+}
```

Comment:
This kind of hammers in the idea that core is more special then the other built-ins. Why do we want that? In my mind built-ins are built-ins. The only special one is local because its not git-based. I wish we could eliminate the special `core` "URL" but that seems not worth the compatibility risks at this point. Core shouldn't be any different from community addons, esphome or music assistant in my mind though.